src/libs/etherlib/node_curl.cpp

```diff
@@ -22,8 +22,9 @@ namespace qblocks {
 
     //-------------------------------------------------------------------------
     CCurlContext::CCurlContext(void) {
+        CToml toml(configPath("quickBlocks.toml"));
         headers      = "Content-Type: application/json\n";
-        baseURL      = "http://localhost:8545";
+        baseURL      = toml.getConfigStr("settings", "rpcEndpoint", "http://localhost:8545");
         callBackFunc = writeCallback;
```

Comment:
Two things:

1) We can use the global pointer to the configuration file which is loaded at the start of every program instead of opening the file again. You can remove the CToml line and replace the getConfigStr call with

    getGlobalConfig()->getConfigStr("settings", "rpcEndpoint", "http://localhost:8545")

2) I like the word 'rpcEndpoint' but let's use the word 'provider' as they do in web3 (here: https://web3js.readthedocs.io/en/1.0/web3.html#setprovider).  So same code but use 'provider' instead of 'rpcEndpoint'.